firefox-ios/Client/Frontend/Browser/SearchViewController.swift

```diff
@@ -238,11 +238,18 @@ class SearchViewController: SiteTableViewController,
         profile.firefoxSuggest?.interruptReader()
 
         let tempSearchQuery = searchQuery
+        var providers = [SuggestionProvider]()
+
+        if includeSponsored && NimbusSuggestFeatureLayer().isAmpSuggestionAvailable() {
+            providers.append(.amp)
+        }
+        if includeNonSponsored && NimbusSuggestFeatureLayer().isWikiSuggestionAvailable() {
+            providers.append(.wikipedia)
+        }
```

Comment:
Then, here, we can do something like:

```suggestion
        let providers = [.amp, .wikipedia]
            .filter { NimbusSuggestFeatureLayer().isSuggestionProviderAvailable($0) }
            .filter {
                switch $0 {
                case .amp: includeSponsored
                case .wikipedia: includeNonSponsored
                default: false
                }
            }
```

(If UniFFI enums supported `CaseIterable`, we could do `SuggestionProvider.allCases` instead of `[.amp, .wikipedia]`).

WDYT?

Reply:
oh this is fancy